eng/Directory.Build.Common.targets

```diff
@@ -198,6 +198,11 @@
 
   <!--TODO: end-->
 
+  <!-- *********** Files needed for Authentication ************** -->
+  <ItemGroup Condition="'$(IncludeAuthenticationSharedSource)' == 'true'">
+    <Compile Include="$(AzureCoreSharedSources)AzureKeyCredentialPolicy.cs" LinkBase="Shared" />
```

Comment:
I'm curious what motivated this change since it doesn't really simplify the inclusion of the policy. It's still a one liner in the csproj either way.

Reply:
I also don't think we should be including it anyway. The class will end up in *every* assembly that includes shared source whether it's used or not. The C# linker doesn't elide types not used - not unless doing AOT with tree shaking.

Reply:
> I'm curious what motivated this change since it doesn't really simplify the inclusion of the policy. It's still a one liner in the csproj either way.

Yes. at first workaround, we can add this policy all the time without a property switch.